homeassistant/components/reolink/__init__.py

```diff
@@ -326,7 +326,19 @@ def migrate_entity_ids(
             else:
                 new_device_id = f"{device_uid[0]}_{host.api.camera_uid(ch)}"
             new_identifiers = {(DOMAIN, new_device_id)}
-            device_reg.async_update_device(device.id, new_identifiers=new_identifiers)
+            existing_device = device_reg.async_get_device(identifiers=new_identifiers)
+            if existing_device is None:
+                device_reg.async_update_device(
+                    device.id, new_identifiers=new_identifiers
+                )
+            else:
+                _LOGGER.error(
```

Comment:
Why is this an error if we are solving it directly in the next line?

Reply:
> There should not have been two duplicate devices in the first place.

This makes no sense. So, you don't know why this is, you communicate an error, next fix it?

Shouldn't this be a warning then?

Reply:
You are right, I have changed it to warnings.

No I don't know why it occurs except for the firmware update -> downgrade -> update (but I don't know of all users having this issue have actually done that). But I indeed do now how to fix it.